File: Original Artifact/app/src/main/java/com/example/freeman_option1/SMSPermissionScreen.java
package com.example.freeman_option1;

import androidx.appcompat.app.AppCompatActivity;
import androidx.core.app.ActivityCompat;
import androidx.core.content.ContextCompat;

import android.Manifest;
import android.content.Intent;
import android.content.pm.PackageManager;
import android.os.Bundle;
import android.view.View;
import android.widget.Toast;

public class SMSPermissionScreen extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_smspermission_screen);

    }

    public void onClickSmsPermission(View v)
    {
        //Switches to the grid view screen once clicked, which prompts user for SMS permission
        startActivity(new Intent(SMSPermissionScreen.this, GridViewLayout.class));
    }
}

File: Data-Structures-and-Algorithms/app_inventory_management/src/main/java/com/example/freeman_option1/SMSPermissionScreen.kt
package com.example.freeman_option1

import android.content.Intent
import android.os.Bundle
import android.view.View
import androidx.annotation.VisibleForTesting
import androidx.appcompat.app.AppCompatActivity

class SMSPermissionScreen : AppCompatActivity() {

    @VisibleForTesting
    public override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_smspermission_screen)
    }

    fun onClickSmsPermission(v: View?) {
        //Switches to the grid view screen once clicked, which prompts user for SMS permission
        startActivity(Intent(this@SMSPermissionScreen, GridViewLayout::class.java))
    }
}